mypy/main.py

```diff
@@ -551,14 +557,6 @@ def add_invertible_flag(flag: str,
     if options.quick_and_dirty:
         options.incremental = True
 
-    # try setting a valid Python executable based on a specified version
-    if options.python_version and not options.python_executable:
-        options.python_executable = _python_executable_from_version(options.python_version)
-
-    # Set Python version if given Python executable, but no version
-    if options.python_executable and not options.python_version:
-        options.python_version = _python_version_from_executable(options.python_executable)
```

Comment:
Why not leave this here?

Reply:
There is a coherence check that process_options doesn't mutate the state of args. I moved it out of there to keep it consistent.